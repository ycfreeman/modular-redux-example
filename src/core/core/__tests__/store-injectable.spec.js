describe("test store", () => {
  const testInitState = {
    test: 0
  };

  const testAction = {
    type: "TEST_ACTION"
  };

  const testAction1 = {
    type: "TEST_ACTION_1"
  };

  const testReducer = (state = testInitState, action) => {
    switch (action.type) {
      case "TEST_ACTION":
        return Object.assign({}, state, { test: state.test + 1 });
      case "TEST_ACTION_1":
        return Object.assign({}, state, { test: state.test - 1 });
      default:
        return Object.assign({}, state);
    }
  };

  let module;

  beforeEach(() => {
    Date.now = jest.fn().mockReturnValue(0);

    module = require("../store"); // anything with plugin_rewire just use dist for sanity
  });

  afterEach(() => {
    module = null;
    jest.resetModules();
    jest.resetAllMocks();
  });

  it("installReducer", () => {
    const { storeCreator, installReducer } = module;

    installReducer("test", testReducer);

    const store = storeCreator();
    expect(store).not.toBeUndefined();
    expect(store.getState()).toMatchSnapshot();
    store.dispatch(testAction);
    expect(store.getState()).toMatchSnapshot();

    installReducer("test2", testReducer);
    store.dispatch(testAction1);
    expect(store.getState()).toMatchSnapshot();
  });
});
