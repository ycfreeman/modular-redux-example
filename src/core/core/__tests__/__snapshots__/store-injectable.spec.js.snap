// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test store installReducer 1`] = `
Object {
  "test": Object {
    "test": 0,
  },
}
`;

exports[`test store installReducer 2`] = `
Object {
  "test": Object {
    "test": 1,
  },
}
`;

exports[`test store installReducer 3`] = `
Object {
  "test": Object {
    "test": 0,
  },
  "test2": Object {
    "test": -1,
  },
}
`;
